### DISCUSSION TITLE
Usage of Set in QueryConfig

### DISCUSSION THREAD

**Discussion by silvanheller** [vitrivr/cineast#171] [General]
The QueryConfig class is used in API Endpoints, which results in OpenAPI generated Code includes `import { Set } from './set';`, which causes compilation errors (which go away after removing the line...). I see the following options:
- Remove the usage of `Set` in QueryConfig parameters exposed to the outside (currently used by `Hint` and `relevantSegmentIDs`)
- Ignore and simply document somewhere when the OpenAPI code needs to be regenerated. This is suboptimal since we might wish to automate the code-generation process someday.
- Fix upstream in openapi-codegen

This is absolutely not urgent, we'll merge https://github.com/vitrivr/vitrivr-ng/pull/54 anyway, but I wanted to leave this here for future reference.

More general, there's an unresolved question about the process by which changes to the Cineast API are made available for vitrivr-ng and in Unity packages

**Comment by lucaro** [comment]
I'd go for the first option. During transmission, the relevant fields are structured as lists anyway and their contents can easily be checked for duplicates afterwards.

I don't understand what you mean by the last sentence.
